sdk/eventhub/Azure.Messaging.EventHubs/src/Amqp/AmqpEventHubProducer.cs

```diff
@@ -0,0 +1,417 @@
+            Argument.AssertNotNull(connectionScope, nameof(connectionScope));
+            Argument.AssertNotNull(messageConverter, nameof(messageConverter));
+            Argument.AssertNotNull(retryPolicy, nameof(retryPolicy));
+
+            EventHubName = eventHubName;
+            PartitionId = partitionId;
+            ConnectionScope = connectionScope;
+            MessageConverter = messageConverter;
+            SendLink = new FaultTolerantAmqpObject<SendingAmqpLink>(timeout => CreateLinkAndEnsureProducerStateAsync(partitionId, timeout, CancellationToken.None) , link => link.SafeClose());
+
+            _retryPolicy = retryPolicy;
+            _tryTimeout = retryPolicy.CalculateTryTimeout(0);
+        }
+
+        /// <summary>
+        ///   Updates the active retry policy for the producer.
+        /// </summary>
+        ///
+        /// <param name="newRetryPolicy">The retry policy to set as active.</param>
+        ///
+        public override void UpdateRetryPolicy(EventHubRetryPolicy newRetryPolicy)
+        {
+            Argument.AssertNotNull(newRetryPolicy, nameof(newRetryPolicy));
+
+            _retryPolicy = newRetryPolicy;
+            _tryTimeout = _retryPolicy.CalculateTryTimeout(0);
+        }
+
+        /// <summary>
+        ///   Sends a set of events to the associated Event Hub using a batched approach.  If the size of events exceed the
+        ///   maximum size of a single batch, an exception will be triggered and the send will fail.
+        /// </summary>
+        ///
+        /// <param name="events">The set of event data to send.</param>
+        /// <param name="sendOptions">The set of options to consider when sending this batch.</param>
+        /// <param name="cancellationToken">An optional <see cref="CancellationToken"/> instance to signal the request to cancel the operation.</param>
+        ///
+        public override async Task SendAsync(IEnumerable<EventData> events,
+                                             SendOptions sendOptions,
+                                             CancellationToken cancellationToken)
+        {
+            Argument.AssertNotNull(events, nameof(events));
+            Argument.AssertNotClosed(_closed, nameof(AmqpEventHubProducer));
+
+            AmqpMessage messageFactory() => MessageConverter.CreateBatchFromEvents(events, sendOptions?.PartitionKey);
+            await SendAsync(messageFactory, sendOptions?.PartitionKey, cancellationToken).ConfigureAwait(false);
+        }
+
+        /// <summary>
+        ///   Sends a set of events to the associated Event Hub using a batched approach.
+        /// </summary>
+        ///
+        /// <param name="eventBatch">The event batch to send.</param>
+        /// <param name="cancellationToken">An optional <see cref="CancellationToken"/> instance to signal the request to cancel the operation.</param>
+        ///
+        /// <returns>A task to be resolved on when the operation has completed.</returns>
```

Comment:
Some methods in this class are missing the `returns` tag (only methods that return a `Task`, it seems).